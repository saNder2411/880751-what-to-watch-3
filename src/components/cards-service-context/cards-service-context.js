import {createContext} from 'react';

const {
  Provider: CardsServiceProvider,
  Consumer: CardsServiceConsumer
} = createContext();

export {CardsServiceProvider, CardsServiceConsumer};
