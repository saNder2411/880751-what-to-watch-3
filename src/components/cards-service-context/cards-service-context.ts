import * as React from 'react';

const {
  Provider: CardsServiceProvider,
  Consumer: CardsServiceConsumer
} = React.createContext(void 0);

export {CardsServiceProvider, CardsServiceConsumer};
