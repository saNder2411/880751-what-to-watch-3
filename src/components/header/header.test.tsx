import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {CardsServiceProvider} from '../cards-service-context/cards-service-context';
import configureStore from 'redux-mock-store';
import {BrowserRouter, Switch, Route} from 'react-router-dom';
import createAPI from '../../api';
import CardsService from '../../services/cards-service';
import thunk from 'redux-thunk';

import Header from './header';
import Logo from '../logo/logo';
import UserBlock from '../user-block/user-block';
import AddReviewBreadcrumbs from '../add-review-breadcrumbs/add-review-breadcrumbs';


const API = createAPI(() => {});
const cardsService = new CardsService(API);

const mockStore = configureStore([thunk]);
const backgroundImage = `bg-the-grand-budapest-hotel`;
const title = `Bohemian Rhapsody`;


describe(`Render Header`, () => {
  it(`Should Header render correctly in CardScreen`, () => {
    const store = mockStore({
      appState: {
        screen: `CARD`,
        selectedCardId: 1,
      },
      user: {
        isAuthorized: false,
        userData: {},
        userDataLoading: true,
        userDataError: null,
      },
    });
    const markup = renderer
      .create(
          <Provider store={store}>
            <CardsServiceProvider value={cardsService}>
              <BrowserRouter>
                <Switch>
                  <Route path='/'>
                    <Header title={title} backgroundImage={backgroundImage}>
                      <Logo />
                      <UserBlock />
                    </Header>
                  </Route>
                </Switch>
              </BrowserRouter>
            </CardsServiceProvider>
          </Provider>
      )
      .toJSON();

    expect(markup).toMatchSnapshot();
  });

  it(`Should Header render correctly in AddReviewScreen`, () => {
    const store = mockStore({
      appState: {
        screen: `ADD_REVIEW`,
        selectedCardId: 1,
      },
      user: {
        isAuthorized: false,
        userData: {},
        userDataLoading: true,
        userDataError: null,
      },
    });
    const markup = renderer
      .create(
          <Provider store={store}>
            <CardsServiceProvider value={cardsService}>
              <BrowserRouter>
                <Switch>
                  <Route path='/'>
                    <Header title={title} backgroundImage={backgroundImage}>
                      <Logo />
                      <UserBlock />
                      <AddReviewBreadcrumbs title={title} />
                    </Header>
                  </Route>
                </Switch>
              </BrowserRouter>
            </CardsServiceProvider>
          </Provider>
      )
      .toJSON();

    expect(markup).toMatchSnapshot();
  });

  it(`Should Header render correctly in Main`, () => {
    const store = mockStore({
      appState: {
        screen: `MAIN`,
        selectedCardId: 1,
      },
      user: {
        isAuthorized: false,
        userData: {},
        userDataLoading: true,
        userDataError: null,
      },
    });
    const markup = renderer
      .create(
          <Provider store={store}>
            <CardsServiceProvider value={cardsService}>
              <BrowserRouter>
                <Switch>
                  <Route path='/'>
                    <Header title={title} backgroundImage={backgroundImage}>
                      <Logo />
                      <UserBlock />
                    </Header>
                  </Route>
                </Switch>
              </BrowserRouter>
            </CardsServiceProvider>
          </Provider>
      )
      .toJSON();

    expect(markup).toMatchSnapshot();
  });
});

