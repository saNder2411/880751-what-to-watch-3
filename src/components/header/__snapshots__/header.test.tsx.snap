// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Should Header render correctly in AddReviewScreen 1`] = `
Array [
  <div
    className="movie-card__bg"
  >
    <img
      alt="Bohemian Rhapsody"
      src="bg-the-grand-budapest-hotel"
    />
  </div>,
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <nav
      className="breadcrumbs"
    >
      <ul
        className="breadcrumbs__list"
      >
        <li
          className="breadcrumbs__item"
        >
          <a
            className="breadcrumbs__link"
            href="/cards/1/"
            onClick={[Function]}
          >
            Bohemian Rhapsody
          </a>
        </li>
        <li
          className="breadcrumbs__item"
        >
          <a
            className="breadcrumbs__link"
          >
            Add review
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login/"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>,
]
`;

exports[`Render Header Should Header render correctly in CardScreen 1`] = `
Array [
  <div
    className="movie-card__bg"
  >
    <img
      alt="Bohemian Rhapsody"
      src="bg-the-grand-budapest-hotel"
    />
  </div>,
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login/"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>,
]
`;

exports[`Render Header Should Header render correctly in Main 1`] = `
Array [
  <div
    className="movie-card__bg"
  >
    <img
      alt="Bohemian Rhapsody"
      src="bg-the-grand-budapest-hotel"
    />
  </div>,
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login/"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>,
]
`;
